tests/acceptance/features/bootstrap/WebUIPasswordPolicyContext.php

```diff
@@ -170,6 +176,53 @@ public function theAdminSetsLowercaseLettersUsingTheWebui(
 		);
 	}
 
+	/**
+	 * @When user :username enters the current password, chooses a new password :newPassword and confirms it using the webUI
+	 *
+	 * @param string $username
+	 * @param string $newPassword
+	 *
+	 * @return void
+	 */
+	public function theUserEntersTheCurrentPasswordChoosesANewPasswordAndConfirmsItUsingTheWebui($username, $newPassword) {
+		$currentPassword = $this->featureContext->getPasswordForUser($username);
+		$this->passwordPolicySettingsPage->chooseNewPassword($currentPassword, $newPassword, $this->getSession());
+	}
+
+	/**
+	 * @param string $username
+	 * @param string $password
+	 *
+	 * @return OwncloudPage
+	 * @throws \Exception
+	 */
+	public function loginWithExpiredPassword($username, $password) {
+		$session = $this->getSession();
+		$this->loginPage->waitTillPageIsLoaded($session);
+		$this->loginPage->loginAs(
+			$username,
+			$password
+		);
+		$this->featureContext->asUser($username);
+	}
+
+	/**
+	 * @When user :username logs in with expired password using the webUI
+	 *
+	 * @param string $username
+	 *
+	 * @return OwncloudPage
+	 * @throws \Exception
+	 */
+	public function userLogsInWithExpiredPasswordUsingTheWebui($username) {
+		$usernameActual = $this->featureContext->getActualUsername($username);
+		$this->webUILoginContext->theUserBrowsesToTheLoginPage();
+		$this->loginWithExpiredPassword(
+			$usernameActual,
+			$this->featureContext->getPasswordForUser($usernameActual)
```

Comment:
```suggestion
			$this->featureContext->getPasswordForUser($username)
```
The actual username is fetched inside this function.

Reply:
@kiranparajuli589 can you elaborate more? Here, I am getting password for a user using the `actual name`.